Sources/NIOCore/IO.swift

```diff
@@ -107,13 +121,46 @@ private func reasonForError(errnoCode: CInt, reason: String) -> String {
     }
 }
 
+#if os(Windows)
+private func reasonForWinError(_ code: DWORD) -> String {
+    let dwFlags: DWORD = DWORD(FORMAT_MESSAGE_ALLOCATE_BUFFER)
+                       | DWORD(FORMAT_MESSAGE_FROM_SYSTEM)
+                       | DWORD(FORMAT_MESSAGE_IGNORE_INSERTS)
+
+    var buffer: UnsafeMutablePointer<WCHAR>?
+    let dwResult: DWORD = withUnsafeMutablePointer(to: &buffer) {
```

Comment:
This code is kinda wild: it allocates a pointer with a value of `nil`, then takes a pointer to that pointer and casts it to pointer to 2 WCHARs, then passes that to `FormatMessageW`, telling it that it should allocate at least zero bytes. This code scares the hell of out of me.

I appreciate that Win32 is the villain here, but can we add a comment explaining that because we've set `FORMAT_MESSAGE_ALLOCATE_BUFFER` that `FormatMessageW` will allocate a buffer for us, but that it requires us to type-pun the input because it only takes an LPWSTR instead of a pointer to one?

Reply:
Sure, I can add a comment, but yes, this is the call in this case.  The fact that it takes a `LPWSTR` here is just mindboggling.